temperature/utils.py

```diff
@@ -0,0 +1,27 @@
+import json
+
+import httpx
+from fastapi import HTTPException
+
+from settings import settings
+
+
+WEATHER_URL = (
+    f"http://api.weatherapi.com/v1/current.json?key={settings.WEATHER_TOKEN}"
+)
+
+
+async def get_temperature(city_name: str) -> float:
+    async with httpx.AsyncClient() as client:
+        response = await client.get(WEATHER_URL + f"&q={city_name}")
+
+        if response.status_code == 200:
+            return response.json()["current"]["temp_c"]
```

Comment:
The code assumes that the response JSON contains a 'current' key with a 'temp_c' subkey. Ensure that this structure matches the actual API response format.